.github/workflows/static-analysis.yml

```
@@ -1,116 +1,117 @@
           restore-keys: ${{ runner.os }}-composer-
 
-      - name: "Install dependencies with Composer"
-        run: "composer install --no-interaction --no-progress --prefer-dist"
+      - name: Install dependencies
+        run: composer install --no-interaction --no-progress --prefer-dist
 
-      - name : "Run PHPStan"
-        run: "phpstan analyse -c phpstan.neon --no-progress --no-interaction --error-format=checkstyle | cs2pr"
+      - name: Run PHPStan
+        run: ./vendor/bin/phpstan analyse -c phpstan.neon --no-progress --no-interaction --error-format=checkstyle | cs2pr
 
   static-analysis-psalm:
-    name: "Static Analysis with Psalm"
+    name: Static Analysis with Psalm
     runs-on: ubuntu-latest
-
-    strategy:
-      matrix:
-        php-version:
-          - "8.0"
     steps:
-      - name: "Checkout"
+      - name: Checkout
         uses: actions/checkout@v2
 
-      - name: "Setup PHP ${{ matrix.php-version }}"
+      - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
-          php-version: ${{ matrix.php-version }}
-          tools: psalm:4.3.1, cs2pr
+          php-version: 8.0
+          tools: cs2pr
           coverage: none
 
-      - name: "Get composer cache directory"
+      - name: Get composer cache directory
         id: composer-cache
         run: echo "::set-output name=dir::$(composer config cache-files-dir)"
 
-      - name: "Cache composer dependencies"
+      - name: Install dependencies
+        run: composer install --no-interaction --no-progress --prefer-dist
+
+      - name: Run Psalm
+        run: ./vendor/bin/psalm --show-info=false --output-format=checkstyle --shepherd | cs2pr
+
+  static-analysis-phpmd:
+    name: Static Analysis with PHPMD
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 7.4
+
+      - name: Get composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Cache dependencies
         uses: actions/cache@v2
         with:
-          path: "${{ steps.composer-cache.outputs.dir }}"
-          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
           restore-keys: ${{ runner.os }}-composer-
 
-      - name: "Install dependencies with Composer"
-        run: "composer install --no-interaction --no-progress --prefer-dist"
+      - name: Install dependencies
+        run: composer install --no-interaction --no-progress --prefer-dist
 
-      - name : "Run Psalm"
-        run: "psalm --output-format=checkstyle | cs2pr"
+      - name: Run PHP Mess Detector
+        run: ./vendor/bin/phpmd src text --exclude src/Annotation ./phpmd.xml
```

`--exclude src/Annotation`はなくて良さそうですかね？

```suggestion
        run: ./vendor/bin/phpmd src text ./phpmd.xml
```